FitGoal/FitGoal/Singup&Singin/SignUpViewController.swift

```diff
@@ -163,6 +168,36 @@ class SignUpViewController: UIViewController, AuthenticationTypeSwitcherViewDele
         show(vc, sender: self)
     }
     
+    func userDidEndEditing(textFieldType: TextFieldType, with text: String) {
```

Comment:
I don't like this function, it seems to repetitive. I would expect it to be something like:

```swift
switch textFieldType {
case .username:
    customAuthenticator.username = text
case .emailAddress:
    customAuthenticator.email = text
...
}

if let errors = customAuthenticator.errors {
    // do something with the errors here.
}
```